tests/search/test_search.py

```diff
@@ -1,106 +1,140 @@
+    def setUp(self):
+        # Role / user
+        role_user, _ = UserRole.objects.get_or_create(role=UserRole.Role.USER)
+        self.user = User.objects.create_user(
+            email="test@example.com",
+            password="password123",
+            role=role_user,
+            is_active=True,
+        )
+        self.client.force_authenticate(user=self.user)
 
+        self.industry = Industry.objects.create(name="Healthcare")
+        self.location = Location.objects.create(country="UA", city="Kyiv")
+        self.category = Category.objects.create(name="Tech")
 
+        # Startup
+        self.startup = Startup.objects.create(
+            user=self.user,
+            company_name="Test Startup",
+            description="AI Startup in healthcare",
+            stage="seed",
+            founded_year=2020,
+            industry=self.industry,
+            location=self.location,
+            email="hello@test-startup.com",
+            team_size=5,
+        )
 
+        changed = False
+        for flag in ("is_active", "is_published", "is_verified", "published", "visible"):
+            if hasattr(self.startup, flag) and not getattr(self.startup, flag):
+                setattr(self.startup, flag, True)
+                changed = True
+        if changed:
+            self.startup.save()
 
+        # Project
+        self.project = Project.objects.create(
+            startup=self.startup,
+            category=self.category,
+            title="Health AI", 
+            description="AI project for healthcare",
+            status="active",   
+            funding_goal=Decimal("10000.00"),
+            current_funding=Decimal("0.00"),
+            email="project@test-startup.com",
+        )
+        changed = False
+        for flag in ("is_active", "is_published", "is_verified", "published", "visible"):
+            if hasattr(self.project, flag) and not getattr(self.project, flag):
+                setattr(self.project, flag, True)
+                changed = True
+        if changed:
+            self.project.save()
 
+    @staticmethod
+    def _make_hit(doc_id):
+        """ES hit supporting both hit.id and hit.meta.id access."""
+        hit = mock.Mock()
+        hit.id = str(doc_id)
+        hit.meta = mock.Mock()
+        hit.meta.id = str(doc_id)
+        return hit
 
+    def _chainable_search(self, search_obj, hits, qs=None):
+        """
+        Prepare a mocked search object compatible with the SUT:
+          - iteration directly over search (for hit in search)
+          - .query/.filter/.source/.extra return the same object
+          - slicing [:N] returns the same object
+          - .execute() also returns iterable(hits)
+          - (optional) .to_queryset() -> qs
+        """
+        for m in ("query", "filter", "source", "extra"):
+            getattr(search_obj, m).return_value = search_obj
+
+        search_obj.__getitem__.return_value = search_obj
+
+        search_obj.__iter__ = lambda s: iter(hits)
```

Comment:
Overriding iter with a lambda on a mock may not work with all mock usages. Use mock.Mock's spec and __iter__ setup for better compatibility.
```
search_obj.__iter__.side_effect = lambda: iter(hits)
```